core/app/[locale]/(default)/webpages/contact/[id]/page.tsx

```diff
@@ -1,54 +1,195 @@
+    id: 'email',
+    name: 'email',
+    label: `${t('emailLabel')} *`,
+    type: 'email',
+    required: true,
+  };
+
+  const commentsField: Field = {
+    id: 'comments',
+    name: 'comments',
+    label: `${t('commentsLabel')} *`,
+    type: 'textarea',
+    required: true,
+  };
+
+  const optionalFields = contactFields
+    .filter((field): field is ContactField => Object.hasOwn(fieldMapping, field))
+    .map<Field>((field) => ({
+      id: field,
+      name: field,
+      label: t(fieldMapping[field]),
+      type: 'text',
+      required: false,
+    }));
+
+  return [
+    ...toGroupsOfTwo([emailField, ...optionalFields]),
+    commentsField,
+    pageIdField,
+    pagePathField,
+  ];
+}
+
+export async function generateMetadata({ params }: Props): Promise<Metadata> {
+  const { id } = await params;
+  const webpage = await getWebPage(id);
   const { pageTitle, metaDescription, metaKeywords } = webpage.seo;
 
   return {
-    title: pageTitle,
+    title: pageTitle || webpage.title,
     description: metaDescription,
-    keywords: metaKeywords,
+    keywords: metaKeywords ? metaKeywords.split(',') : null,
   };
 }
 
-export default async function WebPage({ params }: Props) {
+export default async function ContactPage({ params, searchParams }: Props) {
   const { id } = await params;
+  const { success } = await searchParams;
+  const t = await getTranslations('WebPages.ContactUs.Form');
 
-  const data = await getWebpageData({ id: decodeURIComponent(id) });
-  const webpage = data.node?.__typename === 'ContactPage' ? data.node : null;
+  // TODO: Use reCaptcha
+  // const recaptchaSettings = await bypassReCaptcha(data.site.settings?.reCaptcha);
 
-  if (!webpage) {
-    notFound();
+  if (success === 'true') {
+    return (
+      <WebPageContent
+        breadcrumbs={getWebPageBreadcrumbs(id)}
+        webPage={getWebPageWithSuccessContent(id, t('success'))}
+      >
+        <ButtonLink
+          className="mt-8 @2xl:mt-12 @4xl:mt-16"
+          href="/"
+          size="large"
+          type="submit"
+          variant="primary"
+        >
+          {t('successCta')}
+        </ButtonLink>
+      </WebPageContent>
+    );
   }
 
-  const { name, htmlBody } = webpage;
-  const recaptchaSettings = await bypassReCaptcha(data.site.settings?.reCaptcha);
-
   return (
-    <>
-      <div className="mx-auto mb-10 flex flex-col justify-center gap-8 lg:w-2/3">
-        <h1 className="text-4xl font-black lg:text-5xl">{name}</h1>
-        <div dangerouslySetInnerHTML={{ __html: htmlBody }} />
+    <WebPageContent breadcrumbs={getWebPageBreadcrumbs(id)} webPage={getWebPage(id)}>
+      <div className="mt-8 @2xl:mt-12 @4xl:mt-16">
+        <DynamicForm
+          action={submitContactForm}
+          fields={await getContactFields(id)}
+          submitLabel={t('submitFormText')}
+        />
       </div>
-
-      <ContactUs node={webpage} reCaptchaSettings={recaptchaSettings} />
-    </>
+    </WebPageContent>
```

Comment:
Why not use `DynamicFormSection` here directly?

Reply:
@jorgemoya because the `<SectionLayout>` of the dynamic form section messes with the padding. Since this is a child of the `WebPageContent` component, that would mean we have a section within a section, and by default, `SectionLayout` adds additional x-padding, so when using that it messes up the layout. Also, in this particular case, we don't need the whole section - just the form.

Here's an example of how it looks when we do use the `DynamicFormSection`:
![image](https://github.com/user-attachments/assets/ae0112d6-b431-4af3-8969-ee17f27765ca)

As you can see, the `SectionLayout` within a `SectionLayout` messes it up, and the `SectionLayout` component offers no way to override the inner `div` that adds this padding.

Reply:
@jorgemoya Using the `DynamicForm` by itself allows us to piggy-back off the web page section layout:
![image](https://github.com/user-attachments/assets/23a8c050-4751-456f-af3c-195b3284a112)

Reply:
Ah, this is because the `We're happy to answer...` is the html body from the webpage response?

Reply:
> Ah, this is because the `We're happy to answer...` is the html body from the webpage response?

@jorgemoya Yes, exactly